phpBB/phpbb/controller/helper.php

```diff
@@ -192,4 +243,86 @@ public function get_current_url()
 	{
 		return generate_board_url(true) . $this->request->escape($this->symfony_request->getRequestUri(), true);
 	}
+
+	/**
+	 * Handle display actions for footer, e.g. SQL report and credit line
+	 *
+	 * @param bool $run_cron Flag whether cron should be run
+	 *
+	 * @return void
+	 */
+	public function display_footer($run_cron = true)
+	{
+		$this->display_sql_report();
+
+		$this->template->assign_vars([
+				'DEBUG_OUTPUT'			=> phpbb_generate_debug_output($this->db, $this->config, $this->auth, $this->user, $this->dispatcher),
+				'TRANSLATION_INFO'		=> $this->language->is_set('TRANSLATION_INFO') ? $this->language->lang('TRANSLATION_INFO') : '',
+				'CREDIT_LINE'			=> $this->language->lang('POWERED_BY', '<a href="https://www.phpbb.com/">phpBB</a>&reg; Forum Software &copy; phpBB Limited'),
+
+				'U_ACP'					=> ($this->auth->acl_get('a_') && !empty($this->user->data['is_registered'])) ? append_sid("{$this->admin_path}index.{$this->php_ext}", false, true, $this->user->session_id) : '',
+		]);
+
+		if ($run_cron)
+		{
+			$this->set_cron_task();
+		}
+	}
+
+	/**
+	 * Display SQL report
+	 *
+	 * @return void
+	 */
+	protected function display_sql_report()
+	{
+		if ($this->sql_explain && $this->request->variable('explain', false) && $this->auth->acl_get('a_'))
```

Comment:
This code is now in two places, also still in **functions.php** under the function name `phpbb_check_and_display_sql_report` https://github.com/phpbb/phpbb/blob/8bdad7d73e6c0876110bee033bfa382d16856c1d/phpBB/includes/functions.php#L4319

Maybe let that function call this one or add a deprecation message?

Reply:
Agreed :+1: